test/image/mocks/box_log_scale.json

```diff
@@ -0,0 +1,94 @@
+{
+    "data": [
+        {
+            "y": [
+                1.0,
+                1.1,
+                1.0,
+                0.9,
+                1.2,
+                2.0,
+                1.5,
+                2.3,
+                1.7,
+                2.2,
+                1.0,
+                1.1,
+                1.2,
+                0.9,
+                1.1
+            ],
+            "x": [
+                1,
+                1,
+                1,
+                1,
+                1,
+                10,
+                10,
+                10,
+                10,
+                10,
+                100,
+                100,
+                100,
+                100,
+                100
+            ],
+            "type": "box"
```

Comment:
Have you tried plotting these samples as `type: 'violin'` traces?

Violin and box traces share a lot of the same code, so it might be nice to fix the log-position problems for violin traces in the same go. 

@s417-lama please let us know if you're not interested in working on a fix for `violin` traces. We should be able to find some time to work on it before releasing `v1.51.0`.

Reply:
Oh, I didn't consider violin traces.
I would be glad for you to work on violin plots. I appreciate it.

Reply:
Great. Thanks!! The fix for violins should be very similar than the one for boxes.

Reply:
Fixed! 349ef640517fc2429e8fb9364f7c30925507fddc